pysheds/grid.py

```diff
@@ -299,11 +303,16 @@ def read_raster(self, data, data_name, band=1, window=None, window_crs=None,
                 affine = f.window_transform(ix_window)
                 data = np.squeeze(data)
                 shape = data.shape
+            if mask_geometry:
+                mask = rasterio.features.geometry_mask(mask_geometry, shape, affine, invert=True)
+                if not mask.any():  # no mask was applied if all False, out of bounds
+                    warnings.warn('mask_geometry does not fall within the bounds of the raster!')
+                    mask = ~mask  # return mask to all True and deliver warning???
```

Comment:
Can remove question marks now

Reply:
cool, thanks! Feel free to adjust comments if they can make more sense put another way. I'm happy with how this looks. cheers!